api.planx.uk/saveAndReturn/validateSession.ts

```diff
@@ -62,16 +65,31 @@ export async function validateSession(
           ...currentFlow[key],
         }));
         if (alteredNodes.length) {
+          // each EnrichedCrumb will include a sectionId where relevant (used later)
+          const enrichedAndOrderedBreadcrumbs: Array<EnrichedCrumb> = sortBreadcrumbs(
+            savedFlow,
+            sessionData.data.breadcrumbs
+          );
           const removedBreadcrumbs: Breadcrumb = {};
           alteredNodes.forEach((node) => {
-            // if the session breadcrumbs include any altered content, remove those breadcrumbs so the user will be re-prompted to answer those questions
-            if (
-              sessionData &&
-              Object.keys(sessionData.data.breadcrumbs).includes(node.id!)
-            ) {
+            // if the session breadcrumbs include any altered content,
+            // remove those breadcrumbs so the user will be re-prompted to answer those questions
+            const affectedBreadcrumb = enrichedAndOrderedBreadcrumbs.find(
+              (crumb: EnrichedCrumb) => crumb.id == node.id!
+            );
+            if (affectedBreadcrumb) {
+              // remove affected breadcrumbs
               removedBreadcrumbs[node.id!] =
                 sessionData.data.breadcrumbs[node.id!];
               delete sessionData.data.breadcrumbs[node.id!];
+
+              // also remove the associated section
+              const affectedSectionId = affectedBreadcrumb.sectionId;
+              if (affectedSectionId) {
+                removedBreadcrumbs[affectedSectionId] =
+                  sessionData.data.breadcrumbs[affectedSectionId];
+                delete sessionData.data.breadcrumbs[affectedSectionId];
+              }
```

Comment:
How does this hold up to multiple changes nodes within the same section? I _think_ `sessionData.data.breadcrumbs[affectedSectionId]` will be `undefined` on the second pass.

Reply:
Good catch :+1: